File: scripts/do-flatten-moves.py
#!/usr/bin/env python3
"""Perform Phase A retail monorepo flattening moves."""
import shutil
import os
import sys

def main():
    os.chdir(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
    
    print("Phase A: Flattening retail monorepo...")
    print()
    
    # 1. Move API src
    api_src = 'apps/retail-api/apps/api/src'
    dst_src = 'apps/retail-api/src'
    if os.path.exists(api_src):
        if not os.path.exists(dst_src):
            shutil.move(api_src, dst_src)
            print(f"✓ Moved {api_src} -> {dst_src}")
        else:
            print(f"⚠ {dst_src} already exists, merging...")
            for item in os.listdir(api_src):
                src_item = os.path.join(api_src, item)
                dst_item = os.path.join(dst_src, item)
                if os.path.exists(dst_item):
                    print(f"  ⚠ Skipping {item} (already exists)")
                else:
                    shutil.move(src_item, dst_item)
                    print(f"  ✓ Moved {item}")
    
    # 2. Merge API scripts
    api_scripts = 'apps/retail-api/apps/api/scripts'
    dst_scripts = 'apps/retail-api/scripts'
    if os.path.exists(api_scripts):
        if not os.path.exists(dst_scripts):
            os.makedirs(dst_scripts, exist_ok=True)
        for item in os.listdir(api_scripts):
            src_item = os.path.join(api_scripts, item)
            dst_item = os.path.join(dst_scripts, item)
            if os.path.exists(dst_item):
                print(f"  ⚠ Skipping script {item} (already exists)")
            else:
                shutil.move(src_item, dst_item)
                print(f"  ✓ Moved script {item}")
        try:
            os.rmdir(api_scripts)
        except:
            pass
    
    # 3. Move worker
    worker_src = 'apps/retail-api/apps/worker'
    if os.path.exists(worker_src):
        if not os.path.exists('apps/retail-worker'):
            shutil.move(worker_src, 'apps/retail-worker')
            print(f"✓ Moved {worker_src} -> apps/retail-worker")
        else:
            print("⚠ apps/retail-worker already exists")
    
    # 4. Move web
    web_src = 'apps/retail-api/apps/web'
    if os.path.exists(web_src):
        if not os.path.exists('apps/retail-web-legacy'):
            shutil.move(web_src, 'apps/retail-web-legacy')
            print(f"✓ Moved {web_src} -> apps/retail-web-legacy")
        else:
            print("⚠ apps/retail-web-legacy already exists")
    
    # 5. Clean up empty directories
    try:
        if os.path.exists('apps/retail-api/apps/api'):
            os.rmdir('apps/retail-api/apps/api')
        if os.path.exists('apps/retail-api/apps'):
            os.rmdir('apps/retail-api/apps')
    except:
        pass
    
    print()
    print("✓ Moves complete!")

if __name__ == '__main__':
    main()

File: scripts/do-flatten-moves.js
#!/usr/bin/env node
/** Perform Phase A retail monorepo flattening moves. */
const fs = require('fs');
const path = require('path');

const repoRoot = path.resolve(__dirname, '..');

function moveDir(src, dst) {
  if (!fs.existsSync(src)) {
    console.log(`⚠ Source not found: ${src}`);
    return false;
  }
  if (fs.existsSync(dst)) {
    console.log(`⚠ Destination exists: ${dst}, merging...`);
    const items = fs.readdirSync(src);
    for (const item of items) {
      const srcItem = path.join(src, item);
      const dstItem = path.join(dst, item);
      if (fs.existsSync(dstItem)) {
        console.log(`  ⚠ Skipping ${item} (already exists)`);
      } else {
        fs.renameSync(srcItem, dstItem);
        console.log(`  ✓ Moved ${item}`);
      }
    }
    try {
      fs.rmdirSync(src);
    } catch (e) {
      // Directory not empty, that's ok
    }
    return true;
  } else {
    fs.mkdirSync(path.dirname(dst), { recursive: true });
    fs.renameSync(src, dst);
    console.log(`✓ Moved ${src} -> ${dst}`);
    return true;
  }
}

console.log('Phase A: Flattening retail monorepo...\n');

// 1. Move API src
moveDir(
  path.join(repoRoot, 'apps/retail-api/apps/api/src'),
  path.join(repoRoot, 'apps/retail-api/src')
);

// 2. Merge API scripts
const apiScripts = path.join(repoRoot, 'apps/retail-api/apps/api/scripts');
const dstScripts = path.join(repoRoot, 'apps/retail-api/scripts');
if (fs.existsSync(apiScripts)) {
  if (!fs.existsSync(dstScripts)) {
    fs.mkdirSync(dstScripts, { recursive: true });
  }
  const items = fs.readdirSync(apiScripts);
  for (const item of items) {
    const srcItem = path.join(apiScripts, item);
    const dstItem = path.join(dstScripts, item);
    if (fs.existsSync(dstItem)) {
      console.log(`  ⚠ Skipping script ${item} (already exists)`);
    } else {
      fs.renameSync(srcItem, dstItem);
      console.log(`  ✓ Moved script ${item}`);
    }
  }
  try {
    fs.rmdirSync(apiScripts);
  } catch (e) {}
}

// 3. Move worker
moveDir(
  path.join(repoRoot, 'apps/retail-api/apps/worker'),
  path.join(repoRoot, 'apps/retail-worker')
);

// 4. Move web
moveDir(
  path.join(repoRoot, 'apps/retail-api/apps/web'),
  path.join(repoRoot, 'apps/retail-web-legacy')
);

// 5. Clean up empty directories
try {
  const apiDir = path.join(repoRoot, 'apps/retail-api/apps/api');
  if (fs.existsSync(apiDir)) {
    fs.rmdirSync(apiDir);
  }
  const appsDir = path.join(repoRoot, 'apps/retail-api/apps');
  if (fs.existsSync(appsDir)) {
    fs.rmdirSync(appsDir);
  }
} catch (e) {
  // Not empty or doesn't exist, that's ok
}

console.log('\n✓ Moves complete!');

